azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java

```diff
@@ -290,7 +290,7 @@ protected String baseUri() {
         }
     }
 
-    protected RestClient buildRestClient(RestClientBuilder builder, boolean isMocked) {
+    protected synchronized RestClient buildRestClient(RestClientBuilder builder, boolean isMocked) {
```

Comment:
Would you let us know why it need `synchronized`. This method seems not sharing anything (and almost just a static method)?

Reply:
It call a static function in `RetryPolicy`, which will cause iterator error if not synchronized.